logstash-core/src/main/java/org/logstash/ackedqueue/SettingsImpl.java

```diff
@@ -0,0 +1,246 @@
+        );
+    }
+
+    public static Builder fileSettingsBuilder(final String dirForFiles) {
+        return new BuilderImpl(dirForFiles);
+    }
+
+    public static Builder memorySettingsBuilder() {
+        return memorySettingsBuilder("");
+    }
+
+    public static Builder memorySettingsBuilder(final String dirForFiles) {
+        return new BuilderImpl(dirForFiles).checkpointMaxInterval(0).checkpointMaxAcks(1)
+            .checkpointMaxWrites(1);
+    }
+
+    private SettingsImpl(final String dirForFiles,
+        final CheckpointIOFactory checkpointIOFactory,
+        final PageIOFactory pageIOFactory, final Class elementClass, final int capacity,
+        final long queueMaxBytes, final int maxUnread, final int checkpointMaxAcks,
+        final int checkpointMaxWrites, final int checkpointMaxInterval) {
+        this.dirForFiles = dirForFiles;
+        this.checkpointIOFactory = checkpointIOFactory;
+        this.pageIOFactory = pageIOFactory;
+        this.elementClass = elementClass;
+        this.capacity = capacity;
+        this.queueMaxBytes = queueMaxBytes;
+        this.maxUnread = maxUnread;
+        this.checkpointMaxAcks = checkpointMaxAcks;
+        this.checkpointMaxWrites = checkpointMaxWrites;
+        this.checkpointMaxInterval = checkpointMaxInterval;
+    }
+
+    @Override
+    public int getCheckpointMaxAcks() {
+        return checkpointMaxAcks;
+    }
+
+    @Override
+    public int getCheckpointMaxWrites() {
+        return checkpointMaxWrites;
+    }
+
+    @Override
+    public int getCheckpointMaxInterval() {
+        return checkpointMaxInterval;
+    }
+
+    @Override
+    public CheckpointIOFactory getCheckpointIOFactory() {
+        return checkpointIOFactory;
+    }
+
+    public PageIOFactory getPageIOFactory() {
+        return pageIOFactory;
+    }
+
+    @Override
+    public Class getElementClass()  {
+        return this.elementClass;
+    }
+
+    @Override
+    public String getDirPath() {
+        return dirForFiles;
+    }
+
+    @Override
+    public long getQueueMaxBytes() {
+        return queueMaxBytes;
+    }
+
+    @Override
+    public int getCapacity() {
+        return capacity;
+    }
+
+    @Override
+    public int getMaxUnread() {
+        return this.maxUnread;
+    }
+
+    private static final class BuilderImpl implements Builder {
+
+        private final String dirForFiles;
+
+        private final CheckpointIOFactory checkpointIOFactory;
+
+        private final PageIOFactory pageIOFactory;
+
+        private final Class elementClass;
+
+        private final int capacity;
+
+        private final long queueMaxBytes;
+
+        private final int maxUnread;
+
+        private final int checkpointMaxAcks;
+
+        private final int checkpointMaxWrites;
+
+        private final int checkpointMaxInterval;
+
+        private BuilderImpl(final String dirForFiles) {
+            this(dirForFiles, null, null, null, 0, 0L,
+                0, 1024, 1024, 1000
+            );
```

Comment:
Is there a way to avoid having all these parameters passed in like this - it makes it tricky to understand what all of these parameters mean at a glance, and increases the risk of inadvertent transposition of properties